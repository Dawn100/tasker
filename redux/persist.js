import { AsyncStorage } from "react-native";

export const loadState = async () => {
  try {
    const taskerStore = await AsyncStorage.getItem("TASKERTODOS");
    if (taskerStore !== null) {
      return JSON.parse(taskerStore);
    }

    console.log("load state returned errr-> " + error);
    return undefined;
  } catch (error) {
    console.log("load state error " + error);
    return undefined;
  }
};

export const saveStore = async store => {
  try {
    let storingValue = JSON.stringify({
      todoStore: store,
      time: new Date()
    });
    await AsyncStorage.setItem("TASKERTODOS", storingValue);
  } catch (err) {
    console.log("Save store error" + err);
  }
};
