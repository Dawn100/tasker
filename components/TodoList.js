import React, { Component } from "react";
import { ScrollView } from "react-native";
import Todo from "./Todo";
import { connect } from "react-redux";
import { completeTodo, deleteTodo } from "../redux/actions";

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return (
      <ScrollView>
        {this.props.todos.map((todo, index) => (
          <Todo
            key={index}
            onComplete={()=>{this.props.onComplete(todo.id)}}
            onDelete={()=>{this.props.onDelete(todo.id)}}
            todo={todo}
          />
        ))}
      </ScrollView>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onComplete: id => {
      dispatch(completeTodo(id));
    },
    onDelete: id => {
      dispatch(deleteTodo(id));
    }
  };
};

const mapStateToProps = storeState => {
  return {
    todos: storeState.todos
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList);
