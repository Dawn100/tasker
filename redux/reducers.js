import { combineReducers } from "redux";
import { ADD_TODO, DELETE_TODO } from "./actions";

const todosReducer = (state = [], action) => {
  switch (action.type) {
    case ADD_TODO:
      return [
        {
          todo: action.payload.todo,
          completed: false,
          id: state.reduce((maxId, todo) => Math.max(todo.id, maxId), -1) + 1
        },
        ...state
      ];
    case DELETE_TODO:
      state.map((todo, index) => {
        if (todo.id === action.payload.id) {
          state.splice(index, 1);
        }
      });
      return state;

    default:
      return state;
  }
};

export const reducer = combineReducers({
  todos: todosReducer
});
