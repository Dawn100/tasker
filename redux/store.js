import { createStore, applyMiddleware } from "redux";
import { reducer } from "./reducers";
import { loadState,saveStore } from "./persist";
import { createLogger } from "redux-logger";

const persistedState=loadState()
console.log(persistedState)

const store = createStore(reducer,persistedState.todoStore);

store.subscribe(()=>{
	saveStore(store.getState())
})


export default store