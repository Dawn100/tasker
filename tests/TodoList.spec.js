import { create } from "react-test-renderer";
import React from "react";
import TodoList from "../components/TodoList";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import { addTodo } from "../redux/actions";

describe("TodoList Component", () => {
  store.dispatch(addTodo("Attend Class"));

  const todoList = create(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );

  it("Renders self and sub components", () => {
    const todoListJSON = todoList.toJSON();
    expect(todoListJSON).toMatchSnapshot();
    expect(todoListJSON.type).toEqual("RCTScrollView");
    expect(todoListJSON.children[0].type).toEqual("View");
  });
  it("Todos respond to delete and complete events", () => {
    todoList.root
      .findByProps({ testID: "list" })
      .props.children[0].props.onComplete();
    todoList.root
      .findByProps({ testID: "list" })
      .props.children[0].props.onDelete();
    expect(true).toBe(true);
  });
});
