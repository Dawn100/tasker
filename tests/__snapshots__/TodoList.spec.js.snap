// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component Renders self and sub components 1`] = `
<RCTScrollView
  testID="list"
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ccc",
          "color": "#0f0f0f",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        testID="root"
      >
        <Text
          testID="name"
        >
          Attend Class
        </Text>
        <Text
          testID="status"
        >
          Is complete? 
          false
        </Text>
        <Text
          testID="id"
        >
          0
        </Text>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
